infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/AsyncStorageModeValidator.java

```diff
@@ -0,0 +1,159 @@
+ */
+public class AsyncStorageModeValidator implements WorkspaceAttributeValidator {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AsyncStorageModeValidator.class);
+
+  private final String pvcStrategy;
+  private final boolean allowUserDefinedNamespaces;
+  private final String defaultNamespaceName;
+  private final int runtimesPerUser;
+
+  @Inject
+  public AsyncStorageModeValidator(
+      @Named("che.infra.kubernetes.pvc.strategy") String strategy,
+      @Named("che.infra.kubernetes.namespace.allow_user_defined")
+          boolean allowUserDefinedNamespaces,
+      @Nullable @Named("che.infra.kubernetes.namespace.default") String defaultNamespaceName,
+      @Named("che.limits.user.workspaces.run.count") int runtimesPerUser) {
+
+    pvcStrategy = strategy;
+    this.allowUserDefinedNamespaces = allowUserDefinedNamespaces;
+    this.defaultNamespaceName = defaultNamespaceName;
+    this.runtimesPerUser = runtimesPerUser;
+  }
+
+  @Override
+  public void validate(Map<String, String> attributes) throws ValidationException {
+    if (parseBoolean(attributes.get(ASYNC_PERSIST_ATTRIBUTE))) {
+      isEphemeralAttributeValidation(attributes);
+      pvcStrategyValidation();
+      alowUserDefinedNamespaceValidation();
+      nameSpaceStrategyValidation();
+      runtimesPerUserValidation();
+    }
+  }
+
+  @Override
+  public void validateUpdate(Map<String, String> existing, Map<String, String> update)
+      throws ValidationException {
+    if (parseBoolean(update.get(ASYNC_PERSIST_ATTRIBUTE))) {
+      if (isEphemeral(existing) || isEphemeral(update)) {
+        pvcStrategyValidation();
+        alowUserDefinedNamespaceValidation();
+        nameSpaceStrategyValidation();
+        runtimesPerUserValidation();
+      } else {
+        String message =
+            format(
+                "Workspace configuration not valid: Asynchronous storage available only for NOT persistent storage");
+        LOG.warn(message);
+        throw new ValidationException(message);
+      }
+    }
+  }
+
+  private void isEphemeralAttributeValidation(Map<String, String> attributes)
+      throws ValidationException {
+    if (!isEphemeral(attributes)) {
+      String message =
+          format(
+              "Workspace configuration not valid: Asynchronous storage available only for NOT persistent storage");
+      LOG.warn(message);
+      throw new ValidationException(message);
+    }
+  }
+
+  private void runtimesPerUserValidation() throws ValidationException {
+    if (runtimesPerUser > 1) {
+      String message =
+          format(
+              "Workspace configuration not valid: Asynchronous storage available only if 'che.limits.user.workspaces.run.count' set to 1, but got %s",
+              runtimesPerUser);
+      LOG.warn(message);
+      throw new ValidationException(message);
+    }
+  }
+
+  private void nameSpaceStrategyValidation() throws ValidationException {
+    if (!"<username>-che".equals(defaultNamespaceName)) {
```

Comment:
@vparfonov maybe starts with `<username>-` ?

Reply:
IMO it would make sense to either keep it to strict `<username>-che` or `contains("<username>")`

Reply:
@vparfonov I start to doubt if this config is used on Hosted Che. Is it the default one?

Reply:
Yes, make sense, replace with `contains("<username>")`